x/erc20/keeper/ibc_callbacks.go

```diff
@@ -60,6 +60,9 @@ func (k Keeper) OnRecvPacket(
 	evmParams := k.evmKeeper.GetParams(ctx)
 
 	// if sender == recipient, and is not from an EVM Channel recovery was executed
+	// If we received an IBC from non EVM channel the account should be different
+	// If its the same, users can have their funds stuck since they dont have access
+	// to the same priv key
```

Comment:
```suggestion
	// If we received an IBC message from a non-EVM channel,
	// the sender and receiver accounts should be different.
	//
	// If its the same, users can have their funds stuck since they dont have access
	// to the same priv key.
```